arteria/handlers/arteria_runfolder_handlers.py

```diff
@@ -0,0 +1,156 @@
+            status=200
+        )
+    except Exception as e:
+        set_exceptions(e)
+
+
+@routes.get("/runfolders/next")
+async def get_next_runfolder(request):
+    """
+    Finds unprocessed runfolder (state=ready) and then
+    returns some information about this runfolder.
+    """
+    try:
+        runfolder_cls = list_runfolders(
+            Config()['monitored_directories'],
+            filter_key=lambda r: r.state == State.READY
+        )
+
+        if len(runfolder_cls) > 0:
+            runfolder_dict0 = serialize_runfolder_path(runfolder_cls[0], request)
+
+            return web.json_response(
+                data=runfolder_dict0,
+                status=200
+            )
+        else:
+            raise web.HTTPNoContent(
+                reason="No ready runfolders available."
+            )
+    except Exception as e:
+        set_exceptions(e)
+
+
+@routes.get("/runfolders/pickup")
+async def get_pickup_runfolder(request):
+    """
+    Used to start processing runfolders and also sets the runfolder to PENDING state.
+    """
+    try:
+        runfolder_cls = list_runfolders(
+            Config()['monitored_directories'],
+            filter_key=lambda r: r.state == State.READY
+        )
+
+        if len(runfolder_cls) > 0:
+            runfolder_cls[0].state = State.PENDING
+            runfolder_dict0 = serialize_runfolder_path(runfolder_cls[0], request)
+            return web.json_response(
+                data=runfolder_dict0,
+                status=200
+            )
+        else:
+            raise web.HTTPNoContent(
+                reason="No ready runfolders available."
+            )
+    except Exception as e:
+        set_exceptions(e)
+
+
+@routes.get("/runfolders")
+async def get_all_runfolders(request):
+    """
+    Returns information about all the runfolders that
+    match the state specified (or all runfolders when state
+    is not specified)
+    """
+    try:
+        runfolders = list_runfolders(
+            Config()['monitored_directories'],
+            filter_key=lambda r: r.state == State.READY
+        )
+
+        for runfolder_count, runfolder in enumerate(runfolders):
+            runfolders[runfolder_count] = serialize_runfolder_path(runfolder, request)
+
+        return web.json_response(
+            data={"runfolders": runfolders},
+            status=200
+        )
+    except Exception as e:
+        set_exceptions(e)
+
+
+def get_host_link(request):
+    host = request.url.raw_host
+    link = f"{request.scheme}://{host}/api/1.0"
+    link_path = f"{link}{request.path}"
+
+    return host, link_path
+
+
+def serialize_runfolder_path(runfolder_cls, request):
+    """
+    Get the path uri as web.json_response gives an error when
+    self.path is of type Path
+    """
+    runfolder_dict = runfolder_cls.to_dict()
+    runfolder_dict['path'] = Path(runfolder_dict['path']).as_uri()
+    runfolder_dict['host'], runfolder_dict['link'] = get_host_link(request)

```

Comment:
I missed that the first time, but `link` should be a link to the `path` endpoint in case one wants more info about this runfolder, e.g. ` http://mm-xart002:10800/api/1.0/runfolders/path/data/mm-xart002/runfolders/180718_M04499_0151_000000000-BWFHW` (for all endpoints that return a runfolder).